src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs

```diff
@@ -36,6 +36,18 @@ public class AnonymousBodyParameter : TypedRule<SwaggerParameter>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
+
+        /// <summary>
+        /// When to apply the validation rule, before or after it has been merged as a part of 
+        /// its merged document as specified in the corresponding '.md' file
+        /// By default consider all rules to be applied for After only
+        /// </summary>
+        public override ServiceDefinitionDocumentState ValidationRuleMergeState => ServiceDefinitionDocumentState.Individual;
```

Comment:
the default should be "composed (as described in some other comments) which is also why that is the virtual property's value, and yes, I 100% agree we should only specify *deviations* from that default. I expect "composed" to be correct 90% of the time.